app-doc/heirloom-doctools/heirloom-doctools-160308.ebuild

```diff
@@ -0,0 +1,215 @@
+# Copyright 1999-2009 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Header: /var/cvsroot/gentoo-x86/app-doc/heirloom-doctools/Attic/heirloom-doctools-080407-r2.ebuild,v 1.3 2011/01/20 04:42:43 flameeyes dead $
+# revived from n-t-roff github fork.
+
+EAPI=6
+
+inherit flag-o-matic toolchain-funcs multilib
+
+DESCRIPTION="Classic Unix documentation tools ported from OpenSolaris"
+HOMEPAGE="http://heirloom.sourceforge.net/doctools.html"
+SRC_URI="https://github.com/n-t-roff/heirloom-doctools/releases/download/${PV}/${P}.tar.bz2"
+
+LICENSE="CDDL"
+
+SLOT="0"
+
+KEYWORDS="~amd64 ~x86"
+
+IUSE="cxx"
+
+RDEPEND="!sys-apps/groff"
+DEPEND="sys-devel/flex
+	virtual/yacc"
+
+src_prepare() {
+	# Make sure that C++ code is built with CXXFLAGS and not CFLAGS.
+	find . -name Makefile.mk -exec \
+		sed -i \
+			-e '/(CCC)/s:CFLAGS:CXXFLAGS:' \
+		{} +
+
+	# mpm uses C++, we'll build it explicitly if we really want to
+	sed -i -e 's:mpm:$(MPM):' makefile
+
+	# Monkeypatching dependencies to avoid parallel make failure
+	echo "picl.o: picl.c y.tab.h" >> pic/Makefile.mk
+
+  # Fixing compilation issues
+	sed -i -e 's:e.h:e.h y.tab.h:' eqn/eqn.d/Makefile.mk
+	sed -i -e 's:y.tab.h y.tab.h:y.tab.h:' eqn/eqn.d/Makefile.mk
+
+  sed -i -e 's:e.h:e.h y.tab.h:' eqn/neqn.d/Makefile.mk
+	sed -i -e 's:y.tab.h y.tab.h:y.tab.h:' eqn/neqn.d/Makefile.mk
+
+	# Move some stuff around to better suit our filesystem layout
+  sed -i -e '/INSTALL.*grap\.defines/s:$(LIBDIR):/usr/share/heirloom/doctools:' \
+		grap/Makefile.mk || die
+	sed -i -e '/GRAPDEFINES/s:LIBDIR:"/usr/share/heirloom/doctools":' \
+		grap/main.c || die
+
+	sed -i -e '/INSTALL.*eign/s:$(LIBDIR):/usr/share/heirloom/doctools:' \
+		ptx/Makefile.mk || die
+	sed -i -e '/\/eign/s:\(LIB\|REF\)DIR:"/usr/share/heirloom/doctools":' \
+		ptx/ptx.c refer/mkey3.c || die
+
+	sed -i -e 's:$(LIBDIR)/vgrindefs:/usr/share/heirloom/doctools/vgrindefs:' \
+		vgrind/Makefile.mk || die
+	sed -i -e '/\/vgrindefs/s:LIBDIR:"/usr/share/heirloom/doctools":' \
+		vgrind/vfontedpr.c || die
+
+	sed -i -e 's:$(REFDIR)/papers:/usr/share/heirloom/reftools/papers:g' \
+		refer/Makefile.mk || die
+	sed -i -e '/\/papers\/Ind/s:REFDIR:"/usr/share/heirloom/reftools":' \
+		refer/refer1.c || die
```

Comment:
I could around 15 hacks. This really *deserves* using a patch instead of neverending seds.